lesson16/index.js

```diff
@@ -0,0 +1,128 @@
+// const url = "https://myjson.dit.upm.es/api/bins/hyv9";
+const url = "https://myjson.dit.upm.es/api/bins/d9vx";
+const newsBlock = document.querySelector(".news");
+const newsTabs = document.querySelector(".news_Tabs");
+
+const createNewsTabs = (allData) => {
+  allData.forEach((content, index) => {
+    const li = document.createElement("li");
+    li.classList.add("tab_item");
+    li.id = content.category;
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+
+  const selectedData = allData.find((data) => data.isActive === true);
+  primaryImageWrapper.classList.add("news_Img");
+  const img = document.createElement("img");
+  img.src = selectedData.image.img;
+  primaryImageWrapper.appendChild(img);
+
+  renderNewsList(selectedData);
+  document.getElementById(selectedData.category).classList.add("is-active");
+
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+
+    // タブのindexを取得し、スタイル変更
+    const activeTab = newsTabs.querySelector(".is-active");
+    activeTab.classList.remove("is-active");
+    target.classList.add('is-active');
+
+    if (target.textContent === "news") {
+      const targetData = allData.find((d) => d.category === target.id);
+      setCategoryImage(targetData);
+      renderNewsList(targetData);
+
+    } else if (event.target.textContent === "economy") {
+      const targetData = allData.find((d) => d.category === target.id);
+      setCategoryImage(targetData);
+      renderNewsList(targetData);
+
+    } else if (event.target.textContent === "sports") {
+      const targetData = allData.find((d) => d.category === target.id);
+      setCategoryImage(targetData);
+      renderNewsList(targetData);
+    }
```

Comment:
I don't think conditional branching is necessary.

```suggestion
    const targetData = allData.find((d) => d.category === event.target.id);
    setCategoryImage(targetData);
    renderNewsList(targetData);
```

Reply:
[77ad09e](https://github.com/hippopo-moo/js-handson-work/pull/18/commits/77ad09e5b1575a8ae45496fd16fddd1ca2a60160)

I did not understand it well enough.
Since the data is obtained by matching the category of the event target with the category of the JSON data, there was no need to branch.
I fixed it!